astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -24,211 +30,588 @@ class NDArithmeticMixin(object):
     so that the subclass sees NDData as the main superclass. See
     `~astropy.nddata.NDDataArray` for an example.
     """
-    def _arithmetic(self, operand, propagate_uncertainties, name, operation):
+
+    def arithmetic(self, operation, operand,
+                   uncertainty_correlation=0, meta_kwds_operate=None,
+                   meta_kwds_set=None, handle_uncertainties=True,
+                   handle_mask=True, handle_meta=True, compare_wcs=True):
         """
-        {name} another dataset (``operand``) to this dataset.
+        Base method which calculates the result of the arithmetic operation.
+
+        This class determines the result of the arithmetic operation on the
+        ``data`` with respect to their units and then forwards to other methods
+        to calculate the other properties for the result (like uncertainty).
 
         Parameters
         ----------
-        operand : `~astropy.nddata.NDData`
-            The second operand in the operation a {operator} b
-
-        propagate_uncertainties : bool
-            Whether to propagate uncertainties following the propagation rules
-            defined by the class used for the ``uncertainty`` attribute.
+        operation: `str`
```

Comment:
I couldn't remember what we do so took a peek at the Table docs and couple other places and it looks like current practice is to leave off the double backticks. I don't think it is a big deal to put it in, but you definitely don't want single backticks, I think.

I usually put in the python name for the type when possible, so `str` instead of `string`.